pyproject.toml

```diff
@@ -14,19 +14,32 @@ log_cli_level = "INFO"
 line-length = 99
 target-version = ["py38"]
 
-[tool.isort]
-line_length = 99
-profile = "black"
-
 # Linting tools configuration
-[tool.flake8]
-max-line-length = 99
-max-doc-length = 99
+[tool.ruff]
+line-length = 99
+lint.select = ["E", "W", "F", "C", "N", "D", "I001"]
+lint.extend-ignore = [
+    "D203",
+    "D204",
+    "D213",
+    "D215",
+    "D400",
+    "D404",
+    "D406",
+    "D407",
+    "D408",
+    "D409",
+    "D413",
+]
+lint.ignore = ["E501", "D107"]
+lint.per-file-ignores = {"tests/*" = ["D100","D101","D102","D103","D104"]}
```

Comment:
This is what broke dependabot (`key.subkey`) in the vault charms. I had to change them to be under a `[tool.ruff.lint]` section instead.

I don't know why.

Reply:
Ref: https://github.com/canonical/vault-k8s-operator/pull/188

Reply:
Ohh ok, done